pkg/controllers/cleanup/controller.go

```diff
@@ -0,0 +1,150 @@
+	cpolInformer kyvernov1alpha1informers.ClusterCleanupPolicyInformer,
+	polInformer kyvernov1alpha1informers.CleanupPolicyInformer,
+	cjInformer batchv1informers.CronJobInformer,
+) *Controller {
+	queue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName)
+	c := &Controller{
+		client:      client,
+		cpolLister:  cpolInformer.Lister(),
+		polLister:   polInformer.Lister(),
+		cjLister:    cjInformer.Lister(),
+		queue:       queue,
+		cpolEnqueue: controllerutils.AddDefaultEventHandlersT[*kyvernov1alpha1.ClusterCleanupPolicy](logger, cpolInformer.Informer(), queue),
+		polEnqueue:  controllerutils.AddDefaultEventHandlersT[*kyvernov1alpha1.CleanupPolicy](logger, polInformer.Informer(), queue),
+	}
+	controllerutils.AddEventHandlersT(
+		cjInformer.Informer(),
+		func(n *batchv1.CronJob) { c.enqueueCronJob(n) },
+		func(o *batchv1.CronJob, n *batchv1.CronJob) { c.enqueueCronJob(o) },
+		func(n *batchv1.CronJob) { c.enqueueCronJob(n) },
+	)
+	return c
+}
+
+func (c *Controller) Run(ctx context.Context, workers int) {
+	controllerutils.Run(ctx, logger.V(3), ControllerName, time.Second, c.queue, workers, maxRetries, c.reconcile)
+}
+
+func (c *Controller) enqueueCronJob(n *batchv1.CronJob) {
+	if len(n.OwnerReferences) == 1 {
+		if n.OwnerReferences[0].Kind == "ClusterCleanupPolicy" {
+			cpol := &kyvernov1alpha1.ClusterCleanupPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: n.OwnerReferences[0].Name,
+				},
+			}
+			err := c.cpolEnqueue(cpol)
+			if err != nil {
+				logger.Error(err, "failed to enqueue ClusterCleanupPolicy object", cpol)
+			}
+		} else if n.OwnerReferences[0].Kind == "CleanupPolicy" {
+			pol := &kyvernov1alpha1.CleanupPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      n.OwnerReferences[0].Name,
+					Namespace: n.Namespace,
+				},
+			}
+			err := c.polEnqueue(pol)
+			if err != nil {
+				logger.Error(err, "failed to enqueue CleanupPolicy object", pol)
+			}
+		}
+	}
+}
+
+func (c *Controller) getPolicy(namespace, name string) (kyvernov1alpha1.CleanupPolicyInterface, error) {
+	if namespace == "" {
+		cpolicy, err := c.cpolLister.Get(name)
+		if err != nil {
+			return nil, err
+		}
+		return cpolicy, nil
+	} else {
+		policy, err := c.polLister.CleanupPolicies(namespace).Get(name)
+		if err != nil {
+			return nil, err
+		}
+		return policy, nil
+	}
+}
+
+func (c *Controller) getCronjob(namespace, name string) (*batchv1.CronJob, error) {
+	cj, err := c.cjLister.CronJobs(namespace).Get(name)
+	if err != nil {
+		return nil, err
+	}
+	return cj, nil
+}
+
+func (c *Controller) reconcile(ctx context.Context, logger logr.Logger, key, namespace, name string) error {
+	policy, err := c.getPolicy(namespace, name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		logger.Error(err, "unable to get the policy from policy informer")
+		return err
+	}
+	cronjobNs := namespace
+	if namespace == "" {
+		cronjobNs = config.KyvernoNamespace()
+	}
```

Comment:
@NikhilSharmaWe - can you clarify how the namespace is selected to create the cronjob? For clustercleanuppolicy and cleanuppolicy respectively?

Reply:
For CleanupPolicy, the cronjob namespace is the same as the policy's namespace.
For ClusterCleanupPolicy, the cronjob namespace will be the Kyverno namespace.

Reply:
Thanks Nikhil, that's what I understood.

Not sure if it makes sense to create cronjobs for all policies in the kyverno namespace. It could be confusing to use different namespaces for cleanup policies. This is easy to change, we can update it in the follow-up PRs if needed.

Reply:
Cross namespace owner references are not allowed.

Reply:
In case of a namespaced policy the cronjob has to live in the same namespace.